src/deadline/client/cli/_groups/click_logger.py

```diff
@@ -0,0 +1,39 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import click
+import json
+import typing as t
+
+
+class ClickLogger:
+    """
+    Wrapper around click that is JSON aware. Users can instantiate this as a
+    replacement for using `click.echo`. A helper JSON function is also provided
+    to output JSON.
+    """
+
+    def __init__(self, is_json: bool):
+        self._is_json = is_json
+
+    def echo(
+        self,
+        message: t.Optional[t.Any] = None,
+        file: t.Optional[t.IO[t.Any]] = None,
+        nl: bool = True,
+        err: bool = False,
+        color: t.Optional[bool] = None,
+    ):
+        if not self._is_json:
+            click.echo(message, file, nl, err, color)
+
+    def json(
+        self,
+        message: t.Optional[dict] = None,
+        file: t.Optional[t.IO[t.Any]] = None,
+        nl: bool = True,
+        err: bool = False,
+        color: t.Optional[bool] = None,
+        indent=None,
+    ):
+        if self._is_json:
+            click.echo(json.dumps(obj=message, indent=indent), file, nl, err, color)
```

Comment:
Depending on what `self._is_json` is set to, each of these functions can be a no-op. Is that the desired behavior? or do we want to throw an exception or log an error instead? or maybe try to print the message anyway?

Reply:
Ah - the intention is to allow users to call `.json(message)` or `.echo` without thinking about it. Just log as you see fit. It does not need to raise in either case. 

The output is either "human readable" or "json" only, cannot be both, which is what this class helps to do.

We do not want developers to actively think if a CLI is in JSON mode or plain text mode.